packages/key-management/src/util/mapHardwareSigningData.ts

```diff
@@ -466,19 +464,17 @@ const prepareLedgerOutputs = (outputs: CML.TransactionOutputs, knownAddresses: G
         const multiAssetKeys = scope.manage(multiAsset.keys());
         for (let j = 0; j < multiAssetKeys.len(); j++) {
           const policy = scope.manage(multiAssetKeys.get(j));
-          const assets = scope.manage(multiAsset.get(policy));
+          const assets = scope.manage(multiAsset.get(policy))!;
           const tokens = [];
-          if (assets) {
-            const assetsKeys = scope.manage(assets.keys());
-            for (let k = 0; k < assetsKeys.len(); k++) {
-              const assetName = scope.manage(assetsKeys.get(k));
-              const amount = scope.manage(assets.get(assetName));
-              if (assetName && amount) {
-                tokens.push({
-                  amount: amount.to_str(),
-                  assetNameHex: Buffer.from(assetName.name()).toString('hex')
-                });
-              }
+          const assetsKeys = scope.manage(assets.keys());
```

Comment:
The code is nearly identical to prepareTrezorOutputs, but I think that should be a bigger task of refactoring this code to be more DRY

Reply:
Agree. It's arguable to even be committing this change now, given there's other reasons why this transaction will still fail, but it will at least remove a confusing error, so that's the value the partial fix is adding.